### ISSUE TITLE
\Drupal calls should be avoided in classes, use dependency injection instead

### ISSUE THREAD

**Issue by tkiehne** [jtyocum/drupal8-uwauth#3]
`src/EventSubscriber/UwAuthSubscriber.php` has multiple calls to `\Drupal` static class functions; Drupal coding standards are adamant about invoking these via Dependency Injection. These calls are:

```
\Drupal::currentUser()->isAuthenticated()
\Drupal::service('page_cache_kill_switch')->trigger()
\Drupal::config('uwauth.settings')->get('group.source')
\Drupal::config('uwauth.settings')->get('group.map')
\Drupal::config('uwauth.settings')
```

On initial inspection it looks like we will have to invoke three different services through DI in the class constructor.  I can work on a solution but wanted to document first.

**Comment by fgm** [comment]
@tkiehne I suggest you take a look at my fork https://github.com/fgm/uwauth which has had a lot of changes over this original version. It would be nice if it could be merged back into this repo, actually.

For this specific question, my version has indeed complete DI instead of \Drupal:: calls. https://github.com/fgm/uwauth/blob/genericity/src/EventSubscriber/UwAuthSubscriber.php

**Comment by jtyocum** [comment]
@fgm I'll test your pull request in a dev environment, and see how it works in our environment.

**Comment by fgm** [comment]
FWIW, this has been in use at Carrefour since the PR was sent, so it has seen significant usage.

**Comment by tkiehne** [comment]
Fixed in 2.2
